### PR TITLE
Training PPLs

### PR DESCRIPTION
**Pull request by lennym**
* Rebases #555 to remove the standalone changes, and preserve only the schema modifications
* Add permissible purpose section to granted PPL outputs

### CODE REVIEW

File: client/pages/sections/protocols/section.js [hunk 1]
```diff
@@ -27,15 +29,15 @@ class Section extends PureComponent {
...
+                fields={fields.filter(f => f.show === undefined || f.show(project))}
                 values={values}
                 prefix={prefix}
                 onFieldChange={onFieldChange}
               />
             )
             : (
               <ReviewFields
-                fields={fields}
+                fields={fields.filter(f => f.show === undefined || f.show(project))}
```
Comment (lennym) [inline client/pages/sections/protocols/section.js:40]: This should probably move inside the component and be handled there, but that's probably a thing for another PR.

File: client/pages/sections/protocols/section.js [hunk 2]
```diff
@@ -48,4 +50,4 @@ class Section extends PureComponent {
   }
 }
 
-export default withRouter(Section);
+export default withRouter(connect(({ project }) => ({ project }))(Section));
```
Comment (wheelsandcogs) [inline client/pages/sections/protocols/section.js:53]: holy brackets batman

File: client/pages/sections/protocols/section.js (hunk 1 above)
Comment (lennym) [reply client/pages/sections/protocols/section.js:40]: In fact, it looks like it already _is_ handled in `ReviewFields`

https://github.com/UKHomeOffice/asl-projects/blob/master/client/components/review-fields.js#L10-L18

File: client/pages/sections/protocols/section.js (hunk 2 above)
Comment (lennym) [reply client/pages/sections/protocols/section.js]: Fair. Cleaned up a bit.

**Pull request merged**